drivers/include/vcnl40x0.h

```diff
@@ -0,0 +1,147 @@
+    VCNL40X0_PROXIMITY_RATE_4,                   /**< Proximity measurement rate 4 */
+    VCNL40X0_PROXIMITY_RATE_8,                   /**< Proximity measurement rate 8 */
+    VCNL40X0_PROXIMITY_RATE_16,                  /**< Proximity measurement rate 16 */
+    VCNL40X0_PROXIMITY_RATE_31,                  /**< Proximity measurement rate 31 */
+    VCNL40X0_PROXIMITY_RATE_62,                  /**< Proximity measurement rate 62 */
+    VCNL40X0_PROXIMITY_RATE_125,                 /**< Proximity measurement rate 125 */
+    VCNL40X0_PROXIMITY_RATE_250,                 /**< Proximity measurement rate 250 */
+} vcnl40x0_proximity_rate_t;
+
+/**
+ * @brief Ambient light measurement average
+ */
+typedef enum {
+    VCNL40X0_AMBIENT_AVERAGE_1 = 0,              /**< Ambient average measurement 1 */
+    VCNL40X0_AMBIENT_AVERAGE_2,                  /**< Ambient average measurement 2 */
+    VCNL40X0_AMBIENT_AVERAGE_4,                  /**< Ambient average measurement 4 */
+    VCNL40X0_AMBIENT_AVERAGE_8,                  /**< Ambient average measurement 8 */
+    VCNL40X0_AMBIENT_AVERAGE_16,                 /**< Ambient average measurement 16 */
+    VCNL40X0_AMBIENT_AVERAGE_32,                 /**< Ambient average measurement 32 */
+    VCNL40X0_AMBIENT_AVERAGE_64,                 /**< Ambient average measurement 64 */
+    VCNL40X0_AMBIENT_AVERAGE_128,                /**< Ambient average measurement 128 */
+} vcnl40x0_ambient_avg_t;
+
+/**
+ * @brief Ambient light measurement rate
+ */
+typedef enum {
+    VCNL40X0_AMBIENT_RATE_1 = 0,                 /**< Ambient measurement rate 1 */
+    VCNL40X0_AMBIENT_RATE_2,                     /**< Ambient measurement rate 2 */
+    VCNL40X0_AMBIENT_RATE_3,                     /**< Ambient measurement rate 3 */
+    VCNL40X0_AMBIENT_RATE_4,                     /**< Ambient measurement rate 4 */
+    VCNL40X0_AMBIENT_RATE_5,                     /**< Ambient measurement rate 5 */
+    VCNL40X0_AMBIENT_RATE_6,                     /**< Ambient measurement rate 6 */
+    VCNL40X0_AMBIENT_RATE_8,                     /**< Ambient measurement rate 8 */
+    VCNL40X0_AMBIENT_RATE_10,                    /**< Ambient measurement rate 10 */
+} vcnl40x0_ambient_rate_t;
+
+
+/**
+ * @brief   Status and error return codes
+ */
+enum {
+    VCNL40X0_OK = 0,                             /**< Everything was fine */
+    VCNL40X0_ERR_I2C,                            /**< Error initializing the I2C bus */
+    VCNL40X0_ERR_NODEV                           /**< Error wrong device */
+};
+
+/**
+ * @brief Device initialization parameters
+ */
+typedef struct {
+    i2c_t i2c_dev;                               /**< I2C device which is used */
+    uint8_t i2c_addr;                            /**< Address on I2C bus */
+    uint8_t led_current;                         /**< LED current */
+    vcnl40x0_proximity_rate_t proximity_rate;    /**< Proximity rate */
```

Comment:
I don't how smart all compilers are, but just to make sure I'd rather have these to be `uint8_t` as well and make the `enum`s anonymous.